### PR TITLE
[PM-1380] Add `EncryptedKey` column

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

- Add `EncryptedKey` column to database for all types.
- Add endpoint that allows adding an `EncryptedKey` to an existing device

## Code changes

* **src/Api/Controllers/DevicesController.cs:** Add controller method for accepting an update to a devices encrypted key. 
* **util/Migrator/DbScripts/2023-04-27_00_AddEncryptedKeyToDevice.sql:** Add migration for adding the new column, updating stored procedures to allow for it and refreshing the views so that the column can be retrieved.
* Made changes to all the database projects for adding the new column.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: util/Migrator/DbScripts/2023-04-27_00_AddEncryptedKeyToDevice.sql [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        [RevisionDate] = @RevisionDate,
+        [EncryptedKey] = @EncryptedKey
+    WHERE
+        [Id] = @Id
+END
+GO
+
+IF OBJECT_ID('[dbo].[DeviceView]') IS NOT NULL
+BEGIN
+    EXECUTE sp_refreshsqlmodule N'[dbo].[DeviceView]';
```
Comment (jlf0dev) [inline util/Migrator/DbScripts/2023-04-27_00_AddEncryptedKeyToDevice.sql:84]: Nice, we forget to do this sometimes.

File: src/Api/Models/Request/DeviceEncryptedKeyRequestModel.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Bit.Core.Utilities;
+
+namespace Bit.Api.Models.Request;
+
+public class DeviceEncryptedKeyRequestModel
+{
+    [Required]
+    [EncryptedString]
```
Comment (jlf0dev) [inline src/Api/Models/Request/DeviceEncryptedKeyRequestModel.cs:9]: Do you know where the default is set to validate model states? We have a ton of `ModelState.IsValid` in the Admin project but I'm not sure if those are necessary or not

**Review by jlf0dev — APPROVED** [review]
LGTM

File: src/Api/Models/Request/DeviceEncryptedKeyRequestModel.cs (hunk 2 above)
Comment (justindbaur) [reply src/Api/Models/Request/DeviceEncryptedKeyRequestModel.cs:9]: @jlf0dev I actually had to look into this a bit, according to [this](https://learn.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-7.0#model-state) model validation is automatically on when the controller has `[ApiController]` decorated on it and I expected that to be the case in our controllers but that's not the case actually. 

We get it through applying our [own controller filter](https://github.com/bitwarden/server/blob/5474d3da18d92517e32d4a9e07dab9697eb9cace/src/Api/Utilities/ModelStateValidationFilterAttribute.cs) that checks the model state (through it's inherited class) by itself and returns an error response if it's invalid. That filter is added via this convention that we only add in the `Api` project. 

https://github.com/bitwarden/server/blob/5474d3da18d92517e32d4a9e07dab9697eb9cace/src/Api/Startup.cs#L148

So it seems that the ones in `Admin` are required right now, but there are multiple ways we could make it unnecessary.

**Pull request closed without merging**